Authentication/SignUp/View/SignupView.swift

```diff
@@ -0,0 +1,106 @@
+//
+//  SignupViewType.swift
+//  Authentication
+//
+//  Created by Daniela Riesgo on 3/18/16.
+//  Copyright © 2016 Wolox. All rights reserved.
+//
+
+import Foundation
+
+public protocol SignupViewType: Renderable {
+ 
+    var view: UIView { get }
+    
+    var usernameLabel: UILabel? { get }
+    var usernameTextField: UITextField? { get }
+    var usernameValidationMessageLabel: UILabel? { get }
+    
+    var emailLabel: UILabel? { get }
+    var emailTextField: UITextField { get }
+    var emailValidationMessageLabel: UILabel? { get }
+    
+    var passwordLabel: UILabel? { get }
+    var passwordTextField: UITextField { get }
+    var passwordValidationMessageLabel: UILabel? { get }
+    var passwordVisibilityButton: UIButton? { get }
+    
+    var passwordConfirmLabel: UILabel? { get }
+    var passwordConfirmTextField: UITextField { get }
+    var passwordConfirmValidationMessageLabel: UILabel? { get }
+    var passwordConfirmVisibilityButton: UIButton? { get }
+    
+    var signupButton: UIButton { get }
+    var signupErrorLabel: UILabel? { get }
+    
+    var termsAndServicesButton: UIButton { get }
+    var termsAndServicesLabel: UILabel? { get }
+    
+    var usernameTextFieldValid: Bool { get set }
+    var usernameTextFieldSelected: Bool { get set }
+    var emailTextFieldValid: Bool { get set }
+    var emailTextFieldSelected: Bool { get set }
+    var passwordTextFieldValid: Bool { get set }
+    var passwordTextFieldSelected: Bool { get set }
+    var passwordConfirmationTextFieldValid: Bool { get set }
+    var passwordConfirmationTextFieldSelected: Bool { get set }
+    var signupButtonEnabled: Bool { get set }
+    var signupButtonPressed: Bool { get set }
+    
```

Comment:
The SignUpViewType and LoginViewType share some common characteristics. I propose to do this, what do you think @guidomb @danielaRiesgo ?

```swift
public protocol Renderable {
    
    var view: UIView { get }
    
    func render()
    
}

public protocol AuthenticationFormType {
    
    var emailLabel: UILabel? { get }
    var emailTextField: UITextField { get }
    var emailValidationMessageLabel: UILabel? { get }
    
    var passwordLabel: UILabel? { get }
    var passwordTextField: UITextField { get }
    var passwordValidationMessageLabel: UILabel? { get }
    var passwordVisibilityButton: UIButton? { get }
    
    var usernameLabel: UILabel? { get }
    var usernameTextField: UITextField? { get }
    var usernameValidationMessageLabel: UILabel? { get }
    
    var emailTextFieldValid: Bool { get set }
    var emailTextFieldSelected: Bool { get set }
}

protocol SignUpFormType: AuthenticationFormType {
    
    var passwordConfirmLabel: UILabel? { get }
    var passwordConfirmTextField: UITextField { get }
    var passwordConfirmValidationMessageLabel: UILabel? { get }
    var passwordConfirmVisibilityButton: UIButton? { get }
    
    var signupButton: UIButton { get }
    
    var signupErrorLabel: UILabel? { get }
    
    var termsAndServicesButton: UIButton { get }
    var termsAndServicesLabel: UILabel? { get }
    
    var passwordTextFieldValid: Bool { get set }
    var passwordTextFieldSelected: Bool { get set }
    
    var usernameTextFieldValid: Bool { get set }
    var usernameTextFieldSelected: Bool { get set }
    
    var passwordConfirmationTextFieldValid: Bool { get set }
    var passwordConfirmationTextFieldSelected: Bool { get set }
    
    var signupButtonEnabled: Bool { get set }
    var signupButtonPressed: Bool { get set }
}

public protocol LoginFormType: AuthenticationFormType {
    
    var view: UIView { get }
    
    var logoImageView: UIImageView { get }
    
    var logInButton: UIButton { get }
    var logInErrorLabel: UILabel? { get }
    
    var signUpLabel: UILabel { get }
    var signUpButton: UIButton { get }
    var recoverPasswordButton: UIButton { get }
    
    var logInButtonEnabled: Bool { get set }
    var logInButtonPressed: Bool { get set }
    var showPassword: Bool { get set }
    
}

typealias SignUpViewType = protocol<SignUpFormType, Renderable>

typealias LoginViewType = protocol<LoginFormType, Renderable>
```